bindings/netstandard/ElectionGuard/ElectionGuard.Decryption/Accumulation/AccumulatedSelection.cs

```diff
@@ -0,0 +1,194 @@
+using ElectionGuard.Decryption.Shares;
+using ElectionGuard.Decryption.Extensions;
+using ElectionGuard.Ballot;
+using ElectionGuard.Guardians;
+using ElectionGuard.ElectionSetup.Extensions;
+using ElectionGuard.ElectionSetup;
+
+namespace ElectionGuard.Decryption.Accumulation;
+
+/// <summary>
+/// accumulated collection of all guardian shares for a given selection.
+/// This object is used to compute the final selection value.
+///
+/// This is a mutable class that is filled in through the decrypion process.
+/// </summary>
+public record AccumulatedSelection : DisposableRecordBase, IElectionSelection
+{
+    /// <summary>
+    /// The object id of the selection.
+    /// </summary>
+    public string ObjectId { get; init; }
+
+    /// <summary>
+    /// The sequence order of the selection.
+    /// </summary>
+    public ulong SequenceOrder { get; init; }
+
+    /// <summary>
+    /// The hash of the SelectionDescription.
+    /// </summary>
+    public ElementModQ DescriptionHash { get; init; }
+
+    /// <summary>
+    /// The accumulated value of the selection. `M-bar` in the spec
+    /// </summary>
+    public ElementModP Value { get; private set; }
+
+    /// <summary>
+    /// The accumulated commitment of all guardian commitments.
+    /// </summary>
+    public ElGamalCiphertext Commitment { get; private set; }
+
+    /// <summary>
+    /// The proof that the accumulated value is correct.
+    /// </summary>
+    public ChaumPedersenProof? Proof { get; private set; }
```

Comment:
this classgets constrcuted as aprt of accumulation, then the challenge/response process runs, and if it is successful then the proof is added